src/full_client.toit

```diff
@@ -270,10 +270,11 @@ abstract class DefaultReconnectionStrategyBase implements ReconnectionStrategy:
   Returns whether the broker had a session for this client, otherwise.
   */
   do_connect transport/ActivityMonitoringTransport
+      --reuse_connection/bool=false
       [--reconnect_transport]
       [--send_connect]
       [--receive_connect_ack]:
-    for i := -1; i < attempt_delays_.size; i++:
+    for i := reuse_connection ? -1 : 0; i < attempt_delays_.size; i++:
       if is_closed: return null
       if i >= 0:
         sleep_duration := attempt_delays_[i]
```

Comment:
So we're going to sleep before trying when reuse_connection is true?

Reply:
Yeah. That wasn't a good approach.
Now just reconnect the transport when i == -1 and we shouldn't reuse the connection.